utils/clients/src/Delegate.ts

```diff
@@ -0,0 +1,97 @@
+/*
+  Copyright 2020 Swap Holdings Ltd.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { ethers } from 'ethers'
+import { chainIds, chainNames, MIN_CONFIRMATIONS } from '@airswap/constants'
+
+const DelegateContract = require('@airswap/delegate/build/contracts/Delegate.json')
+
+export class Delegate {
+  locator: string
+  chainId: string
+  contract: any
+
+  constructor(
+    locator: string,
+    chainId = chainIds.RINKEBY,
+    signerOrProvider?: ethers.Signer | ethers.providers.Provider
+  ) {
+    this.locator = locator
+    this.chainId = chainId
+    this.contract = new ethers.Contract(
+      locator,
+      DelegateContract.abi,
+      signerOrProvider ||
+        ethers.getDefaultProvider(chainNames[chainId].toLowerCase())
+    )
+  }
+
+  async getWallet(): Promise<string> {
+    return await this.contract.tradeWallet()
+  }
+
+  async getMaxQuote(signerToken: string, senderToken: string): Promise<any> {
+    const result = await this.contract.getMaxQuote(signerToken, senderToken)
+    return {
+      signer: {
+        token: signerToken,
+        amount: result.signerAmount.toString(),
+      },
+      sender: {
+        token: senderToken,
+        amount: result.senderAmount.toString(),
+      },
+    }
+  }
```

Comment:
So `getMaxQuote` on the delegate doesn't factor in the balance of the tradeWallet, so to get an accurate `maxQuote` you will need to: 
1. check if the tradeWallet has sufficient balance to cover the rule
2. If not, determine the price from the rule
3. use the tradeWallet balance * price to determine the actual `maxQuote`

Example code for doing that is here: https://github.com/airswap/AirSwap.js/blob/develop/src/delegate/index.js#L126